amx/server/events.lua

```diff
@@ -157,8 +172,11 @@ function putPlayerInClassSelection(player)
 	if not isElement(player) then
 		return
 	end
-	toggleAllControls(player, false, true, false)
 	local playerID = getElemID(player)
+	if g_Players[playerID].doingclasssel then
+		return
+	end

```

Comment:
If we're spectating don't send an event to show the UI. I haven't checked, but I believe this is what SA-MP does and it mirrors my gamemode's behavior in SA-MP.

```suggestion
	if g_Players[playerID].doingclasssel or g_Players[playerID].state == PLAYER_STATE_SPECTATING then
		return
	end
```